### PR TITLE
Remove non-required references from test projects

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Azure.Messaging.EventHubs.Tests.csproj [hunk 1]
```diff
@@ -12,12 +12,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Identity" />
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Azure.Messaging.EventHubs.Tests.csproj:15]: Good catch.   With the common test environment base now holding the credential, looks like this is no longer necessary.

**Review by jsquire — APPROVED** [review]
The Event Hubs and Service Bus parts look good to me.

**Review by MiYanni — APPROVED** [review]

**Review by heaths — APPROVED** [review]
Key Vault and Search changes LGTM

**Review by maririos — APPROVED** [review]
FR and TA LGTM.

**Comment by maririos** [comment]
Hmmm could you also delete this reference in App config? https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/tests/Azure.Data.AppConfiguration.Tests.csproj#L16

**Review by christothes — APPROVED** [review]
👍 Tables

**Review by kinelski — APPROVED** [review]
Metrics Advisor LGTM.

**Pull request merged**